connection.go

```diff
@@ -897,6 +917,14 @@ func (c *Connection) input(p []byte) error {
 		return ErrorConnIsClosed
 	}
 
+	if c.state == StateClosing {
+		c.log(logTypeConnection, "Discarding packet while closing (closePacket=%v)", c.closePacket != nil)
+		if c.closePacket != nil {
+			c.transport.Send(c.closePacket)
+		}
```

Comment:
When wouldn't you want to send something? Shouldn't you generate a new close packet in that case?

Reply:
This is what the spec permits.  You could generate a new packet, and in this case it would be more space efficient, if not computationally efficient, but I wanted to have minq show the permitted behaviour.

You don't send when you received a close, only when you initiate one.  There's a failure mode here where both sides send close packets constantly, with neither one looking at the contents.

Note that this means that you don't look at packets, which means that you can still end up in that nasty failure mode, but only if closes are simultaneous.  It is possible to send the packet less often so that the rate slows over time, but that takes more code.

Reply:
Sorry, I think my question wasn't clear, or maybe I just didn't understand your answer.

Right now, whenever you call sendQueuedFrames() and you're in closing, you send the closePacket, but only if you have one cached. My question is why you don't make one when you don't have one cached.

Reply:
The desired behaviour is to send closePacket in three cases:

1. when you close
2. when you receive a packet, but only if you were the one to initiate close (as in 1 above)
3. when you receive a close, but only the first time

Whether closePacket is nil is used at point 2 to determine whether you are closing because of 1 or 3.

I don't think that I understand the sendQueuedFrames() interaction here.  I'll look into that.

Reply:
Oh, I see my error.  I should exit CheckTimer sooner if the state is closing.  That's a bug.